Helper/DeletedProductSerializer.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Metrilo\Analytics\Helper;
+
+class DeletedProductSerializer extends \Magento\Framework\App\Helper\AbstractHelper
+{
+    public function serialize($deletedProductOrders) {
+        $productBatch = [];
+        foreach ($deletedProductOrders as $order) {
+            
+            foreach ($order->getAllItems() as $item) {
+                $parentProduct  = '';
+                $productOptions = [];
+                
+                $parentItemId   = $item->getParentItemId();
+                $itemId         = $item->getProductId();
+                $itemSku        = $item->getSku();
+                $itemName       = $item->getname();
+                
+                if ($item->getProductType() == 'configurable' || $this->checkForProductIdIndex($itemId, $productBatch) !== false) {
```

Comment:
What does `$this->checkForProductIdIndex($itemId, $productBatch) !== false` mean?

Reply:
Maybe it will be more understandable if it is in a named method?

Reply:
done.